docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+                }
+            )
+        ),
+    #...
+```
+
+This next code snippet shows a slightly more advanced version of the `insert_editor_js` hook usage which is set up to easily append additional scripts for future controllers.
+
+```python
+# wagtail_hooks.py
+from django.utils.html import format_html_join
+from django.templatetags.static import static
+
+from wagtail import hooks
+
+
+@hooks.register('insert_editor_js')
+def editor_js():
+    # add more controller code as needed
+    js_files = ['js/word-count-controller.js',]
+    return format_html_join('\n', '<script src="{0}"></script>',
+        ((static(filename),) for filename in js_files)
+    )
+```
+
+You should be able to see that on your Blog Page, all introduction fields will now have a small `output` element showing the count and max words being used.
+
+#### A word count controller example with a build system
+
+You can import the base controller from the global Stimulus application or if you have a build tooling set up you can install `@hotwired/stimulus` using `npm install @hotwired/stimulus --save`.
+
+```{warning}
+Usage of `class ... extends` from the globally provided `window.Stimulus.use.Controller` cannot be done outside if your code bundles to ES5.
+You will need to use the package provided `Controller` as a base in this case, or update your build output to be ES6/ES2015 or above.
+```
+
+```javascript
+// myapp/static/js/word-count-controller.js
+// const { Controller } = window.Stimulus.use; // if you do not want to use an alias or import your package
+import { Controller } from '@hotwired/stimulus';
+
+class WordCountController extends Controller {
+    // ... the same as above
+}
+
+window.Stimulus.register('word-count', WordCountController);
+```
+
+-   How you import the script will depend on your build solution, but a similar `wagtail_hooks.py` approach can be used.
+-   Using the data attributes in your `FieldPanel`'s `widget` will work exactly the same.
+
+#### Additional tips for bundling
+
+You may want to avoid bundling Stimulus with your JavaScript output, you will need to look at how your build system can support this. You may also need to ensure your target bundle is ES2015 or above to be able to correctly extend the global base Controller class.
+
+For bundler-specific handling of external dependencies or aliasing, see the following links.
+
+-   [Vite library mode](https://vitejs.dev/guide/build.html#library-mode), which uses Rollup configuration
+-   [Rollup external](https://rollupjs.org/configuration-options/#external) and [Rollup output globals](https://rollupjs.org/configuration-options/#output-globals)
+-   [Webpack externals](https://webpack.js.org/configuration/externals/)
+-   [Parcel aliases](https://parceljs.org/features/dependency-resolution/#aliases)
+
+### Using an existing Stimulus controller in HTML
+
+```{warning}
+While many Stimulus controllers are in use, this doesn't imply stable and documented usage is supported for all.
```

Comment:
This – and below "Any built-in admin Stimulus Controller can be used via the Stimulus data attributes." –  is the kinds of statements I don’t see working too well for us. I’d rather we specifically list which controllers should be available for reuse, document those, and keep their API stable.

Reply:
I will just remove this whole section, I am not sure it's worth pushing this for the initial documentation. I agree that we need to be careful here with what we document and based on other PRs it's more likely we will expose (and document) a Django widget, Panel or even template tag before we document too many 'low level' implementation items like this.